mountpoint-s3/src/data_cache/multilevel_cache.rs

```diff
@@ -0,0 +1,325 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use futures::task::{Spawn, SpawnExt};
+use tracing::{trace, warn};
+
+use crate::object::ObjectId;
+
+use super::{BlockIndex, ChecksummedBytes, DataCache, DataCacheResult};
+
+/// A data cache which uses both the local disk and S3 Express One Zone bucket as a storage.
```

Comment:
I would simplify and just stick to only the first sentence.

If you want to provide more detail, you need to go all the way in and also mention how `get_block` can populate the local cache for example. But my preference would be to document the specific methods.

In any case, I would focus on the local vs shared distinction rather than assumptions on relative speed.